src/models/UserModel.ts

```diff
@@ -0,0 +1,30 @@
+import mongoose from "mongoose";
+
+interface IUser {
+  name: string;
+  password: string;
+}
+
+interface userModelInterface extends mongoose.Model<UserDoc> {
+  build(attr: IUser): UserDoc;
+}
+
+interface UserDoc extends mongoose.Document {
+  name: string;
+  password: string;
+}
+
+const userSchema = new mongoose.Schema({
+  name: {
+    type: String,
+    required: true,
```

Comment:
you can add validation in schema for duplication in user name. so no need to check user is present or not mongoose will give you error.